giraffe/src/transforms/band.ts

```diff
@@ -126,6 +125,77 @@ export const getBandName = (fillColumnMap: ColumnGroupMap): string => {
   }, '')
 }
 
+export const alignMinMaxWithBand = (
+  lineData: LineData,
+  bandIndexMap: BandIndexMap
+): LineData => {
+  const {rowIndices, maxIndices, minIndices} = bandIndexMap
+
+  for (let i = 0; i < rowIndices.length; i += 1) {
+    const bandId = rowIndices[i]
+    const minId = minIndices[i]
+    const maxId = maxIndices[i]
+
+    let bandXs = []
```

Comment:
decoupling declaration and assignment by hoisting these lets out of the loop will help reduce memory usage on large datasets

Reply:
+1: in general, we should measure our ui performance before optimizing, but it's a pretty good bet that not creating and destroying memory every iteration of the loop will be more performant than creating it once before the start of the loop.

how big of a loop is this?